main.go

```diff
@@ -183,6 +203,20 @@ func main() {
 	}
 	// +kubebuilder:scaffold:builder
 
+	leaderElectionErr := make(chan error)
+	leaderElectionContext, cancelLeaderElection := context.WithCancel(context.Background())
+	go func() {
+		setupLog.Info("starting leader election manager")
+		leaderElectionErr <- leaderElectionMgr.Start(leaderElectionContext)
+	}()
+
+	select {
+	case <-leaderElectionMgr.Elected():
+	case err := <-leaderElectionErr:
+		setupLog.Error(err, "Leader Election Manager error")
+		os.Exit(1)
+	}
+
 	// Create a default SriovNetworkNodePolicy
 	err = createDefaultPolicy(kubeClient)
```

Comment:
cancelLeaderElection before exit(1) in createDefaultPolicy?

Reply:
done